CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -154,37 +154,42 @@ function Staff:tick()
   self:updateSpeed()
 end
 
-function Staff:checkIfWaitedTooLong()
-  if self.quitting_in then
-    self.quitting_in = self.quitting_in - 1
-    if self.quitting_in < 0 then
-      local rise_windows = self.world.ui:getWindows(UIStaffRise)
-      local staff_rise_window = nil
+function Staff:checkIfWaitedTooLongForRaise()
+  if self.quitting_in >= 0 then return end
+  self.quitting_in = self.quitting_in - 1
```

Comment:
Didn't get the idea here.
But If you mean to swap them

Then
The thing is that `:checkIfWaitedTooLong` is called every tick

`function Staff:tick() 
...
self:checkIfWaitedTooLong()
...
end`

So
`if self.quitting_in >= 0`
Plays here the guard/gate/constraint role for internal block in this flow.

There no any guard/gate/constraint in `:tick()` for this flow.

And if decrement will be outside of this internal block, then that mean it will works every tick.

But I think we don't want it.

We want decrement (timer) only work when deadline is setup.

I think we've messed something up here. 😅

And we should back 
  `if self.quitting_in then`
As a guard/gate/constraint.
Not
  `self.quitting_in >= 0`

Reply:
Apologies. To put it more simply.

Line 158 runs a check if `quitting_in` is over 0 we exit early (nothing else happens in this function).
Line 159 is the decrement of `quitting_in` by 1.

In essence, by having 158 first, `self.quitting_in` never decreases, because your if statement will always be true.